tests/pipelines/stable_diffusion/test_stable_diffusion_inpaint.py

```diff
@@ -252,6 +252,43 @@ def test_stable_diffusion_inpaint_strength_zero_test(self):
         with self.assertRaises(ValueError):
             sd_pipe(**inputs).images
 
+    def test_stable_diffusion_inpaint_mask_latents(self):
+        device = "cpu"
+        components = self.get_dummy_components()
+        sd_pipe = self.pipeline_class(**components).to(device)
+        sd_pipe.set_progress_bar_config(disable=None)
+
+        # normal mask + normal image
+        ##  `image`: pil, `mask_image``: pil, `masked_image_latents``: None
+        inputs = self.get_dummy_inputs(device)
+        inputs["strength"] = 0.9
+        out_0 = sd_pipe(**inputs).images
+
+        # image latents + mask latents
+        inputs = self.get_dummy_inputs(device)
+        image = sd_pipe.image_processor.preprocess(inputs["image"]).to(sd_pipe.device)
+        mask = sd_pipe.mask_processor.preprocess(inputs["mask_image"]).to(sd_pipe.device)
+        masked_image = image * (mask < 0.5)
+
+        generator = torch.Generator(device=device).manual_seed(0)
+        image_latents = (
+            sd_pipe.vae.encode(image).latent_dist.sample(generator=generator) * sd_pipe.vae.config.scaling_factor
+        )
+        torch.randn((1, 4, 32, 32), generator=generator)
```

Comment:
@patrickvonplaten 

Because the `vae.encode()` is a nondeterministic operation, to completely match the result, I had to force the generator to be used in same sequence in both cases, i.e. encode_initial_image -> create noise to add to it -> encode masked_image. For example, here I had to just use it for the sake of using it just so that the `mask_latents` we created here matches what's in the pipeline. 

I feel pretty stupid doing it this way :pleading_face: is there a better method? also, I think this test will easily fail if we make any changes to the pipeline that change the order in how generator used

Reply:
Totally fine for me ! Nice find with the `generator` class & how it works :-)